src/prof.c

```diff
@@ -1818,6 +1825,9 @@ prof_mdump(tsd_t *tsd, const char *filename) {
 	if (!opt_prof || !prof_booted) {
 		return true;
 	}
+	if (tsd->reentrancy_level > 0) {
```

Comment:
For this to trigger, the application would have to make a reentrant manual dump call, which is way outside supported behavior. An assertion should be sufficient.